R/zchunk_LB113.bio_Yield_R_GLU.R

```diff
@@ -25,53 +29,77 @@ module_aglu_LB113.bio_Yield_R_GLU_DISABLED <- function(command, ...) {
 
     all_data <- list(...)[[1]]
 
+    value <- HA_ha <- HA <- Yield_avg <- Yield <- Ratio <- iso <- GCAM_region_ID <- GCAM_REGION_ID
+    GLU <- Ratio_weight <- . <- YieldIndex <- GTAP_crop <- NULL # silence package check notes
+
     # Load required inputs
     iso_GCAM_regID <- get_data(all_data, "common/iso_GCAM_regID")
     L100.LDS_ag_HA_ha <- get_data(all_data, "L100.LDS_ag_HA_ha")
     L100.LDS_ag_prod_t <- get_data(all_data, "L100.LDS_ag_prod_t")
 
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Calculate global average yields for each FAO crop in the base year (31-39 old file)
+    L100.LDS_ag_HA_ha %>%
+      group_by(GTAP_crop) %>%
+      summarise(HA_ha = sum(value)) ->
+      L113.ag_HA_ha_glbl_crop
+
+    L100.LDS_ag_prod_t %>%
+      group_by(GTAP_crop) %>%
+      summarise(value = sum(value)) %>%
+      left_join_error_no_match(L113.ag_HA_ha_glbl_crop, by = "GTAP_crop") %>%
+      mutate(Yield_avg = value / HA_ha) ->
+      L113.ag_prod_t_glbl_crop
+
+    # Calculate each region / zone / crop's comparative yield (41-50)
+    L100.LDS_ag_HA_ha %>%
+      rename(HA = value) %>%
+      left_join_error_no_match(L100.LDS_ag_prod_t, by = c("iso", "GLU", "GTAP_crop")) %>%
+      mutate(Yield = value / HA) %>%
+      # Drop the missing values, where the harvested area was above the min threshold but production was not
+      na.omit ->
+      LDS_ag_Yield_tha
+
+    # Match in the global avg yield for each crop, and compute the ratio from that yield (52-61)
```

Comment:
also aggregating HA and Ratio_weight to GCAM region in this pipeline. might be worth a brief additional comment.